File: FinanceApp.Application/DTOs/ParcelaDTO.cs
using FinanceApp.Domain.Enums;
using TypeGen.Core.TypeAnnotations;

namespace FinanceApp.Application.DTOs
{
    [ExportTsInterface]
    public class ParcelaDTO
    {
        public Guid Id { get; set; }
        public Guid ContaId { get; set; }
        public int NumeroParcela { get; set; }
        public int TotalParcelas { get; set; }
        public decimal ValorParcela { get; set; }
        public DateTime DataVencimento { get; set; }
        public string? Observacao { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: Finance.Web/backend/dto/parcela.dto.ts
// DTOs de parcela baseados no ParcelaController e ParcelaDTO
import { ApiResponse } from './common.dto'

export interface Parcela {
  id: string // Guid do backend convertido para string
  contaId: string // Guid do backend convertido para string
  numeroParcela: number
  totalParcelas: number
  valorParcela: number
  dataVencimento: string
  observacao?: string
  createdAt: string
  updatedAt?: string
}

export interface CreateParcelaRequest {
  contaId: string // Guid do backend convertido para string
  numeroParcela: number
  totalParcelas: number
  valorParcela: number
  dataVencimento: string
  observacao?: string
}

export interface UpdateParcelaRequest {
  valorParcela?: number
  dataVencimento?: string
  observacao?: string
}

// Respostas da API usando ApiResponse genérico
export interface ParcelaResponse extends ApiResponse<Parcela> {}
export interface ParcelasResponse extends ApiResponse<Parcela[]> {}
export interface DeleteParcelaResponse extends ApiResponse<null> {}
